app/screens/search/search.js

```diff
@@ -118,30 +116,35 @@ export default class Search extends PureComponent {
         }, 150);
     }
 
-    componentDidUpdate(prevProps) {
-        const {searchingStatus: status, recent, enableDateSuggestion} = this.props;
+    componentDidUpdate(prevProps, prevState) {
+        const {searchingStatus: status, enableDateSuggestion} = this.props;
+        const {recent} = this.state;
         const {searchingStatus: prevStatus} = prevProps;
         const shouldScroll = prevStatus !== status &&
-            (status === RequestStatus.SUCCESS || status === RequestStatus.STARTED) &&
-            !this.props.isSearchGettingMore && !prevProps.isSearchGettingMore;
+            (status === RequestStatus.SUCCESS || status === RequestStatus.FAILURE) &&
+            !this.props.isSearchGettingMore && !prevProps.isSearchGettingMore && prevState.recent.length === recent.length;
 
         if (this.props.isLandscape !== prevProps.isLandscape) {
             this.refs.searchBar.blur();
         }
 
         if (shouldScroll) {
             setTimeout(() => {
-                const recentLabelsHeight = (recent.length + (Platform.OS === 'ios' ? 1 : 0)) * RECENT_LABEL_HEIGHT;
-                const recentSeparatorsHeight = (recent.length + (Platform.OS === 'ios' ? 0 : 2)) * RECENT_SEPARATOR_HEIGHT;
+                const recentLabelsHeight = recent.length * RECENT_LABEL_HEIGHT;
```

Comment:
Could we use [`SectionList.scrollToLocation`](https://facebook.github.io/react-native/docs/sectionlist#scrolltolocation) for this instead of doing the math ourselves?

Reply:
sadly no, this needs the getItemLayout and that overcomplicate things